core/shapes.cpp

```diff
@@ -428,12 +435,43 @@ void BoardShape::setupInserts() {
 
 const GCodeWriter BoardShape::generateBaseCutout(const Machine& machine) {
   auto tool = machine.tool(machine.baseCutoutTool());
-  auto paths = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  
+  // 1. Get path assuming full wrap of metal edge
+  auto fullWrapPath = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  assert(fullWrapPath.size() == 1);
+  
+  // 2. Form square outlining limit of metal edges (whole board if no partial edges)
+  MCFixed noseEdgeX = 0;
+  if (m_noseEdgeExt != 999.0) {
+    noseEdgeX += (m_noseLength - m_noseEdgeExt);
+  }
+  MCFixed tailEdgeX = m_noseLength + m_effectiveEdge + m_tailLength;
+  if (m_tailEdgeExt != 999.0) {
+    tailEdgeX -= (m_tailLength - m_tailEdgeExt);
+  }
+  Path edgeLimitSquare;
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, -(m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, (m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, (m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  
+  // 3. Clip square
+  auto clippedEdgeSquare = PathUtils::ClipPathsDifference(
+    vector<Path> {edgeLimitSquare} , fullWrapPath);
+  assert(clippedEdgeSquare.size() == 2);
```

Comment:
If you allow a full wrap nose and a partial wrap tail, then this won't be true. I believe you'll end up with one path, like a big wide 'c'.